pkg/serviceaccount/claims.go

```diff
@@ -46,11 +46,11 @@ type privateClaims struct {
 }
 
 type kubernetes struct {
-	Namespace string          `json:"namespace,omitempty"`
-	Svcacct   ref             `json:"serviceaccount,omitempty"`
-	Pod       *ref            `json:"pod,omitempty"`
-	Secret    *ref            `json:"secret,omitempty"`
-	WarnAfter jwt.NumericDate `json:"warnafter,omitempty"`
+	Namespace string           `json:"namespace,omitempty"`
+	Svcacct   ref              `json:"serviceaccount,omitempty"`
+	Pod       *ref             `json:"pod,omitempty"`
+	Secret    *ref             `json:"secret,omitempty"`
+	WarnAfter *jwt.NumericDate `json:"warnafter,omitempty"`
```

Comment:
does this change what we serialize into claims for this token?

Reply:
No, this change only affects Go code but not the serialized representation of the token: In case `WarnAfter` is not set, the corresponding JSON field `warnafter` is not defined. Changing the `WarnAfter` field to a pointer type does not change the original behavour of omitting the field when it is not specified. Similarly, the marshalled representation of the ẀarnAfter` field with a specified value does not change either since its type remains the same but with a pointer.